sdk/communication/communication-short-codes/src/shortCodesClient.ts

```diff
@@ -0,0 +1,192 @@
+} from "@azure/communication-common";
+import { isTokenCredential, KeyCredential, TokenCredential } from "@azure/core-auth";
+import {
+  PipelineOptions,
+  InternalPipelineOptions,
+  createPipelineFromOptions,
+  RestResponse
+} from "@azure/core-http";
+import { PagedAsyncIterableIterator } from "@azure/core-paging";
+import { logger, createSpan, SDK_VERSION } from "./utils";
+import { ShortCodesClient as ShortCodesGeneratedClient } from "./generated/src";
+import { ShortCodes as GeneratedClient } from "./generated/src/operations";
+import {
+  ShortCode,
+  ShortCodesGetShortCodesOptionalParams,
+  ShortCodesUpsertUSProgramBriefOptionalParams,
+  USProgramBrief
+} from "./generated/src/models/";
+import { SpanStatusCode } from "@azure/core-tracing";
+import {
+  DeleteUSProgramBriefOptions,
+  GetUSProgramBriefOptions,
+  SubmitUSProgramBriefOptions
+} from "./models";
+
+/**
+ * Client options used to configure the ShortCodesClient API requests.
+ */
+export interface ShortCodesClientOptions extends PipelineOptions {}
+
+const isShortCodesClientOptions = (options: any): options is ShortCodesClientOptions =>
+  options && !isKeyCredential(options) && !isTokenCredential(options);
+
+export class ShortCodesClient {
+  /**
+   * A reference to the auto-generated ShortCodes HTTP client.
+   */
+  private readonly client: GeneratedClient;
+
+  public constructor(connectionString: string, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: KeyCredential, options?: ShortCodesClientOptions);
+
+  public constructor(url: string, credential: TokenCredential, options?: ShortCodesClientOptions);
+
+  public constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?: KeyCredential | TokenCredential | ShortCodesClientOptions,
+    maybeOptions: ShortCodesClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isShortCodesClientOptions(credentialOrOptions)
+      ? credentialOrOptions
+      : maybeOptions;
+    const libInfo = `azsdk-js-communication-short-codes/${SDK_VERSION}`;
+
+    if (!options.userAgentOptions) {
+      options.userAgentOptions = {};
+    }
+
+    if (options.userAgentOptions.userAgentPrefix) {
+      options.userAgentOptions.userAgentPrefix = `${options.userAgentOptions.userAgentPrefix} ${libInfo}`;
+    } else {
+      options.userAgentOptions.userAgentPrefix = libInfo;
+    }
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info
+        }
+      }
+    };
+
+    const authPolicy = createCommunicationAuthPolicy(credential);
+    const pipeline = createPipelineFromOptions(internalPipelineOptions, authPolicy);
+    this.client = new ShortCodesGeneratedClient(url, pipeline).shortCodes;
+  }
+
+  public listShortCodes(
+    options: ShortCodesGetShortCodesOptionalParams = {}
```

Comment:
Can we introduce a type `ListShortCodesOptions` and use that instead?
Same for the other client functions.

Reply:
Is this the correct way? creating interface that extends the corresponding options class in the models